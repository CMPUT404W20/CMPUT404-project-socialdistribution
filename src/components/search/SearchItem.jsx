import React, { Component } from "react";
import "../../styles/friends-notices-search/SearchItem.scss";
import PropTypes from "prop-types";
import { Link } from "react-router-dom";
import Row from "react-bootstrap/Row";
import Col from "react-bootstrap/Col";

class SearchItem extends Component {
  constructor(props) {
    super(props);
    this.props = props;
  }

  render() {
    const {
      username, userID, host,
    } = this.props;
    const currentHost = localStorage.getItem("host");
    return (
      <Row className="wrapper">
        <Col md={12} lg={4} className="leftColumn">
          <p className="username-link">
            {username}
          </p>
        </Col>
        <Col md={12} lg={3} className="middleColumn">
          <div className="type-wrapper">{ host === currentHost ? "Local" : "Remote"}</div>
        </Col>
        <Col md={12} lg={5} className="rightColumn">
          <div className="button-wrapper">
            <Link
              className="view-profile-button"
              to={{
                pathname: `/profile/${username}`,
                state: { user: { displayName: username, id: userID, host } },
              }}
            >
              View Profile
            </Link>
          </div>
        </Col>
      </Row>
    );
  }
}

SearchItem.propTypes = {
  username: PropTypes.string.isRequired,
  userID: PropTypes.string.isRequired,
  host: PropTypes.string.isRequired,
};

export default SearchItem;
