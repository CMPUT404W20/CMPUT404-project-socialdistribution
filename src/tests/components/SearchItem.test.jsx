import React from "react";
import { shallow } from "enzyme";
import SearchItem from "../../components/search/SearchItem";

describe("Search Item Component", () => {
  it("should render correctly", () => {
    const component = shallow(<SearchItem userID="" username="" host="" />);
    expect(component).toMatchSnapshot();
  });
});
