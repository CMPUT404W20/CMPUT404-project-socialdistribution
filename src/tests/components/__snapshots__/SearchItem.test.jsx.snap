// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Item Component should render correctly 1`] = `
<Row
  className="wrapper"
  noGutters={false}
>
  <Col
    className="leftColumn"
    lg={4}
    md={12}
  >
    <p
      className="username-link"
    />
  </Col>
  <Col
    className="middleColumn"
    lg={3}
    md={12}
  >
    <div
      className="type-wrapper"
    >
      Remote
    </div>
  </Col>
  <Col
    className="rightColumn"
    lg={5}
    md={12}
  >
    <div
      className="button-wrapper"
    >
      <Link
        className="view-profile-button"
        to={
          Object {
            "pathname": "/profile/",
            "state": Object {
              "user": Object {
                "displayName": "",
                "host": "",
                "id": "",
              },
            },
          }
        }
      >
        View Profile
      </Link>
    </div>
  </Col>
</Row>
`;
